esp-storage/src/buffer.rs

```diff
@@ -0,0 +1,61 @@
+use core::{mem::MaybeUninit, slice};
+
+#[cfg(feature = "nor-flash")]
+pub fn uninit_slice(bytes: &[u8]) -> &[MaybeUninit<u8>] {
+    unsafe { core::mem::transmute(bytes) }
+}
+
+#[cfg(feature = "nor-flash")]
+pub fn uninit_slice_mut(bytes: &mut [u8]) -> &mut [MaybeUninit<u8>] {
+    unsafe { core::mem::transmute(bytes) }
+}
+
+pub type FlashWordBuffer = FlashBuffer<4, 1>;
+pub type FlashSectorBuffer = FlashBuffer<4096, 1024>;
+
+#[repr(C)]
+pub union FlashBuffer<const N: usize, const M: usize> {
+    bytes: [MaybeUninit<u8>; M],
+    words: [MaybeUninit<u32>; N],
+}
+
+impl<const N: usize, const M: usize> FlashBuffer<N, M> {
+    pub const fn uninit() -> Self {
+        debug_assert!(N * 4 == M);
```

Comment:
ideally we could turn this into a const-assert

Reply:
would adding a dependency on `static_assertions` just for this be ok?

Reply:
I think there is a way to do it without an additional dependency and keeping the MSRV - we do something similar here: https://github.com/esp-rs/esp-hal/blob/f1ec4a2ac9aa6dbffbaa241c4227804f86fa2a41/esp-wifi/src/lib.rs#L119-L129

Reply:
This technique doesn't work with type parameters.

Reply:
Something like this works
```rust
struct Foo<const N: usize, const M: usize> {
    a: [u8; N],
    b: [u8; M],
}

impl<const N: usize, const M: usize> Foo<N,M> {
    const _CHECK: () = {
        assert!(N == M, "Not good")
    };

    const fn new() -> Self {
        let _ = Self::_CHECK;

        Self {
            a: [0u8; N],
            b: [0u8; M],
        }
    }
}
```